spinoso-time/src/time/chrono/error.rs

```diff
@@ -0,0 +1,72 @@
+use core::fmt;
+use std::error::Error;
+
+/// Error returned when constructing a [`Time`] from a [`ToA`].
+///
+/// This error is returned when a time component in the `ToA` exceeds the maximum
+/// permissible value for a datetime. For example, invalid values include a
+/// datetime 5000 days or 301 seconds.
+///
+/// # Examples
+///
+/// Invalid date component:
+///
+/// ```
+/// # use spinoso_time::chrono::{Offset, Time, ToA, ComponentOutOfRangeError};
+/// let to_a = ToA {
+///     sec: 21,
+///     min: 3,
+///     hour: 23,
+///     day: 5000,
+///     month: 4,
+///     year: 2020,
+///     wday: 0,
+///     yday: 96,
+///     isdst: true,
+///     zone: Offset::Local,
+/// };
+/// let time = Time::try_from(to_a);
+/// assert_eq!(time, Err(ComponentOutOfRangeError::Date));
+/// ```
+///
+/// Invalid time component:
+///
+/// ```
+/// # use spinoso_time::chrono::{Offset, Time, ToA, ComponentOutOfRangeError};
+/// let to_a = ToA {
+///     sec: 301,
+///     min: 3,
+///     hour: 23,
+///     day: 5,
+///     month: 4,
+///     year: 2020,
+///     wday: 0,
+///     yday: 96,
+///     isdst: true,
+///     zone: Offset::Local,
+/// };
+/// let time = Time::try_from(to_a);
+/// assert_eq!(time, Err(ComponentOutOfRangeError::Time));
+/// ```
+///
+/// [`Time`]: [`time::chrono::Time`]
+/// [`ToA`]: [`time::chrono::ToA`]
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[allow(clippy::module_name_repetitions)]
+pub enum ComponentOutOfRangeError {
```

Comment:
ahh I see it moved to the `time::chrono` module. Why are we not sharing this among backends? I don't think it is tied to the `chrono` implementation.

In practice, `artichoke-backend` implements `RubyException` for "foreign" error types from the spinoso crates, so having unified error types which are shared among all backends makes those downstream trait impls a bit easier to write.

Reply:
Ok I took a peek at https://github.com/b-n/artichoke/pull/171 and this makes much more sense. I retract my concern 😅

Reply:
As comment from `lib.rs`, I'm sure there are some commonalities in the errors. Creating an issue :smile: